### DISCUSSION TITLE
Allow custom dotenv path for @remix-run/dev

### DISCUSSION THREAD

**Discussion by anqueue** [remix-run/remix#3893] [Proposals]
Currently, the only location a `.env` file can be for @remix-run/dev is the root directory of the project. Dotenv supports custom paths which is useful for any projects with shared `.env` files amongst other applications however Remix does not let you modify this.

My proposed change is to allow for a custom path option in `remix.config.js` so projects can take advantage of a shared `.env` instead of copy pasting or creating a symlink.

**Comment by kiliman** [comment]
If you need to support different `.env` files, you can install `dotenv-cli` then update *package.json* to add `dotenv -e path/to/.env remix dev`

If you need any configuration beyond the default provided by Remix App Server, you should switch to the Express adapter, which gives you full control. I have a script that lets you eject from RAS to Express https://rmx.fyi/ejectras
